lib-backend-wasm/src/func.rs

```diff
@@ -1309,16 +1523,66 @@ fn encode_appl<H: HeapManager>(
                                 wasmgen::ValType::I32,
                                 wasmgen::ValType::I32,
                             ]),
-                            encode_result(Some(ir::VarType::Any), ctx.options.wasm_multi_value),
+                            // When !ctx.options.wasm_tail_call, functions always return
+                            // i32(is_tail)
+                            encode_result(Some(ir::VarType::Boolean), ctx.options.wasm_multi_value),
                         )),
                     wasmgen::TableIdx { idx: 0 },
                 );
             });
+        }
+
+        // Run post appl calling convention when ctx.options.wasm_tail_call. Else
+        // encode trampolining until ir_return on the stack.
+        if ctx.options.wasm_tail_call {
+            // net wasm stack: [vartype] -> [ir_return_type]
+            encode_post_appl_calling_conv(
+                Some(ir::VarType::Any),
+                ctx.options.wasm_multi_value,
+                ctx.stackptr,
+                mutctx.scratch_mut(),
+                expr_builder,
+            );
```

Comment:
We should not be doing this if `tail_call == true`, because code after a return_call wasm instruction will never be executed.